python/semantic_kernel/connectors/ai/open_ai/services/azure_chat_completion.py

```diff
@@ -2,14 +2,23 @@
 
 
 from logging import Logger
-from typing import Optional
+from typing import TYPE_CHECKING, List, Optional, Tuple, Union
 
+from azure.core.credentials import TokenCredential
+
+from semantic_kernel.connectors.ai.chat_request_settings import ChatRequestSettings
+from semantic_kernel.connectors.ai.open_ai.services.azure_credentials_mixin import (
+    AzureCredentialMixin,
+)
 from semantic_kernel.connectors.ai.open_ai.services.open_ai_chat_completion import (
     OpenAIChatCompletion,
 )
 
+if TYPE_CHECKING:
+    pass
```

Comment:
I don't understand this change. 

What's the use of keeping a this IF with only a "pass"?

The idea of the conditional import was that azure credentials is an "optional" dependency, and so it is not necessarily installed in the target system, if Azure OAI with tokens is used, for instance.
Using the type annotation as string also allowed it to run without the library installed if you were not using Azure Credentials.

With this change I fear that it won't be backwards compatible, and it will be a breaking change as a new dependency is mandatory even if new feature is not used